### PR TITLE
Add batching helper function

### PR DESCRIPTION
**Pull request by nyanhp**
I've recently enjoyed your beautiful module, and was missing batching. One thing is still somewhat missing:  Even with batching, the token lifetime can expire. I have yet to find a good method to get a refresh token if the time expires. I've thought about add-membering the ExpiresAfter property to the token.

Also, I fixed the loop with DeviceCode auth. The exception you're matching in my case contained a whitespace character...

```powershell
Connect-GraphClientSecret -ClientID $ClientId -ClientSecret $ClientSecret -TenantID $TenantId
Set-GraphEndpoint -Type beta
$servicePrincipals = [System.Collections.ArrayList]::new()
$query = "servicePrincipals?&`$filter=tags/any(t:t eq 'WindowsAzureActiveDirectoryIntegratedApp')"
$servicePrincipals.AddRange([array](Invoke-GraphRequest -Query $query))

$araCounter = 1
$idToSp = @{}
$appRoleAssignmentsRequest = foreach ($sp in $servicePrincipals)
{
    @{
        url    = "/servicePrincipals/$($sp.id)/appRoleAssignments"
        method = "GET"
        id     = $araCounter
    }
    $idToSp[$araCounter] = $sp
    $araCounter++
}
$result = Invoke-GraphRequestBatch -Request $appRoleAssignmentsRequest
```

The resulting data in my case is mapped to the IDs - you get the meaning.

### CODE REVIEW

**Review by FriedrichWeinmann — CHANGES_REQUESTED** [review]
Heya @nyanhp,
thanks for the PR and catching the DeviceCode error :)

Looking good mostly, but some issues on the batching command

**Comment by FriedrichWeinmann** [comment]
> I have yet to find a good method to get a refresh token if the time expires

Automatic Reconnect before token expiration is in the works (soon(tm))

File: MiniGraph/functions/Invoke-GraphRequestBatch.ps1 [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        [Parameter(Mandatory)]
+        [hashtable[]]
+        $Request
+    )
+
+    $batchSize = 20 # Currently hardcoded API limit
+    $counter = [pscustomobject] @{ Value = 0 }
+    $batches = $Request | Group-Object -Property { [math]::Floor($counter.Value++ / $batchSize) } -AsHashTable
+
+    $batchResult = [System.Collections.ArrayList]::new()
```
[the remark this answers is not in the document]
Comment (nyanhp) [reply MiniGraph/functions/Invoke-GraphRequestBatch.ps1:38]: Not really, no. The (crude) design evolved and that looks like a remnant of that refactoring

**Review by FriedrichWeinmann — APPROVED** [review]

**Comment by FriedrichWeinmann** [comment]
Alright, merging this now and doing adjustments myself - need to get the next version out today and I definitely want this command in it.
Thanks again for the awesome update and sorry about my impatience and taking over :)

**Comment by FriedrichWeinmann** [comment]
Oh, and the reconnect is implemented as well, so the version coming today is going to have all of the goodies in one package :)

File: MiniGraph/functions/Invoke-GraphRequestBatch.ps1 [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    foreach ($batch in ($batches.GetEnumerator() | Sort-Object -Property Key))
+    {
+        [array] $innerResult = try
+        {
+            $jsonbody = @{requests = [array]$batch.Value } | ConvertTo-Json -Depth 42 -Compress
+            (Invoke-GraphRequest -Query '$batch' -Method Post -Body $jsonbody -ErrorAction Stop).responses
+        }
+        catch [Microsoft.PowerShell.Commands.HttpResponseException]
```
[the remark this answers is not in the document]
Comment (nyanhp) [reply MiniGraph/functions/Invoke-GraphRequestBatch.ps1:47]: Removed

File: MiniGraph/functions/Invoke-GraphRequestBatch.ps1 [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            {
+                continue
+            }
+
+            try
+            {
+                $retriedResults = [array](Invoke-GraphRequestBatch -Name $Name -Request $retry -NoProgress -ErrorAction Stop).responses
+                $null = $batchResult.AddRange($retriedResults)
+            }
+            catch [Microsoft.PowerShell.Commands.HttpResponseException]
```
[the remark this answers is not in the document]
Comment (nyanhp) [reply MiniGraph/functions/Invoke-GraphRequestBatch.ps1:79]: Removed

**Pull request merged**